File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec4iUniformValue.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.joml.Vector4i;
import org.joml.Vector4ic;

public class Vec4iUniformValue extends Vec4iUniform implements UniformValue<Vector4ic> {

    private final String name;

    private final Vector4i value = new Vector4i();

    public Vec4iUniformValue(String name) {
        this.name = name;
    }

    @Override
    public final Vector4ic getUniformValue() {
        return getValue();
    }

    @Override
    public final String getName() {
        return this.name;
    }

    @Override
    public final Vector4i getValue() {
        return this.value;
    }

    @Override
    public final void setValue(Vector4ic value) {
        getValue().set(value);
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec4iUniformValue.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.joml.Vector4i
import org.joml.Vector4ic
import org.saar.maths.JomlDelegates

class Vec4iUniformValue(
    override val name: String,
    initialValue: Vector4ic = Vector4i(0, 0, 0, 0)
) : Vec4iUniform() {
    override var value: Vector4i by JomlDelegates.CachedVector4i(Vector4i(initialValue))
}